src/fu-plugin.c

```diff
@@ -742,6 +771,24 @@ fu_plugin_runner_update_cleanup (FuPlugin *plugin, FuDevice *device, GError **er
 	return TRUE;
 }
 
+void
+fu_plugin_runner_device_register (FuPlugin *plugin, FuDevice *device)
+{
+	FuPluginPrivate *priv = GET_PRIVATE (plugin);
+	FuPluginDeviceRegisterFunc func = NULL;
+
+	/* not enabled */
+	if (!priv->enabled)
+		return;
+
+	/* optional */
+	g_module_symbol (priv->module, "fu_plugin_device_added", (gpointer *) &func);
```

Comment:
I think that registering is still pretty confusing.  Registered is better though if you're happy with that.